In [4]:
from Utils import *
db = VectorDatabase()
import os
os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "max_split_size_mb:4000"
db.initialize_process(chunk_size=256 ,chunk_overlap=200)
from Utils import *
retriever = Retriever()
retriever.process_questions(method='Vector')

evaluator = Evaluation()

evaluator.output_evaluation()

= VectorDatabase initialized =
  - loading data into ChromaDB 
     - collection faq deleted.
     - loading faq done.
     - collection insurance deleted.
     - loading insurance done.
     - collection finance deleted.
     - loading finance done.
= VectorDatabase initialized =
= Retriever initialized =
= Ground truth accuracy =
Retrieval accuracy: 85.33%
Category: insurance, Accuracy: 88.00%
Category: finance, Accuracy: 72.00%
Category: faq, Accuracy: 96.00%


In [1]:
from Utils import *
retriever = Retriever()
retriever.process_questions(method='BM25_Vector')

evaluator = Evaluation()

evaluator.output_evaluation()

resource module not available on Windows
= VectorDatabase initialized =


Building prefix dict from the default dictionary ...
Loading model from cache C:\Users\marks\AppData\Local\Temp\jieba.cache


= Retriever initialized =


Loading model cost 0.629 seconds.
Prefix dict has been built successfully.


= Ground truth accuracy =
Retrieval accuracy: 79.33%
Category: insurance, Accuracy: 88.00%
Category: finance, Accuracy: 58.00%
Category: faq, Accuracy: 92.00%


In [ ]:
from Utils import *
db = VectorDatabase()
import os
os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "max_split_size_mb:4000"
db.initialize_process(chunk_size=400 ,chunk_overlap=300)
from Utils import *
retriever = Retriever()
retriever.process_questions(method='Vector')

evaluator = Evaluation()

evaluator.output_evaluation()

In [ ]:
from Utils import *
db = VectorDatabase()
import os
os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "max_split_size_mb:4000"
db.initialize_process(chunk_size=400 ,chunk_overlap=300)
from Utils import *
retriever = Retriever()
retriever.process_questions(method='Vector')

evaluator = Evaluation()

evaluator.output_evaluation()

In [1]:
import jieba
from typing import List

def chinese_tokenizer(text: str) -> List[str]:
    # Use jieba to segment Chinese text
    return list(jieba.cut(text))

In [ ]:
from llama_index.retrievers.bm25 import BM25Retriever
from llama_index.core import Document
from llama_index.core.node_parser import SentenceSplitter
from llama_index.core import VectorStoreIndex
from llama_index.core.response.notebook_utils import display_source_node
from llama_index.embeddings.huggingface import HuggingFaceEmbedding
import pickle
import chromadb
from llama_index.vector_stores.chroma import ChromaVectorStore  
from typing import List

# Initialize embedding model
my_embedding = HuggingFaceEmbedding(
    model_name="TencentBAC/Conan-embedding-v1"  # Options: "TencentBAC/Conan-embedding-v1", "sensenova/piccolo-base-zh"
)

documents = [Document(text="床前明月光"),
             Document(text="疑是地上霜"),
             Document(text="舉頭望明月"),
             Document(text="低頭思故鄉")]

from llama_index.core import SimpleDirectoryReader
from llama_index.core import VectorStoreIndex, Document, StorageContext
# # load documents
# with open('./reference/insurance.pkl', 'rb') as file:
#     corpus_dict = pickle.load(file)

from Utils import *    
db = VectorDatabase()
vector_index = db.get_vector_index('faq')

                
# documents = []
# for id, text in corpus_dict.items():
#     doc = Document(
#         text=text, 
#         id_=f"doc_id_{id}",
#         metadata={"category": 'insurance', "pid": id}
#     )
#     documents.append(doc)
# from llama_index.core.storage.docstore import SimpleDocumentStore
# docstore = SimpleDocumentStore()

# splitter = SentenceSplitter(chunk_size=256)
# chroma_collection = chromadb.PersistentClient('./chroma').get_or_create_collection('test')
# vector_store =ChromaVectorStore(chroma_collection=chroma_collection)
# nodes = splitter.get_nodes_from_documents(documents)
# docstore.add_documents(nodes)
# vector_index = VectorStoreIndex(
#             nodes=nodes,
#             embed_model=my_embedding,
#             storage_context=StorageContext.from_defaults(vector_store=vector_store)
#         )

# StorageContext.from_defaults(docstore=docstore).docstore.persist('./docstore/test.json')

# docstore = SimpleDocumentStore.from_persist_path('./docstore/test.json')
# storage_context = StorageContext.from_defaults(docstore=docstore, vector_store=vector_store)
# vector_index = VectorStoreIndex(
#     nodes=[], 
#     storage_context=storage_context,
#     embed_model=my_embedding
#     )

def chinese_tokenizer(text: str) -> List[str]:
        # Use jieba to segment Chinese text
        return list(jieba.cut(text))

retriever = BM25Retriever.from_defaults(
    docstore=vector_index.docstore,
    similarity_top_k=2,
    tokenizer=chinese_tokenizer,
    
)

nodes = retriever.retrieve("故乡")

nodes[0].score

The tokenizer parameter is deprecated and will be removed in a future release. Use a stemmer from PyStemmer instead.


= VectorDatabase initialized =


0.0

256 100 
  
= Ground truth accuracy =  

Retrieval accuracy: 84.67% 

Category: insurance, Accuracy: 90.00% 

Category: finance, Accuracy: 68.00% 

Category: faq, Accuracy: 96.00% 


256 200 

= Ground truth accuracy = 

Retrieval accuracy: 86.00% 

Category: insurance, Accuracy: 88.00% 

Category: finance, Accuracy: 74.00% 

Category: faq, Accuracy: 96.00% 



512 256  

= Ground truth accuracy = 

Retrieval accuracy: 83.33% 

Category: insurance, Accuracy: 84.00% 

Category: finance, Accuracy: 70.00% 

Category: faq, Accuracy: 96.00% 


128 64

Retrieval accuracy: 79.33% 

Category: insurance, Accuracy: 86.00% 

Category: finance, Accuracy: 56.00% 

Category: faq, Accuracy: 96.00%

512 400 

= VectorDatabase initialized = 

= Retriever initialized = 

= Ground truth accuracy = 

Retrieval accuracy: 84.67% 

Category: insurance, Accuracy: 88.00% 

Category: finance, Accuracy: 70.00% 

Category: faq, Accuracy: 96.00%

512 450 
 
= VectorDatabase initialized = 

= Retriever initialized = 

= Ground truth accuracy = 

Retrieval accuracy: 83.33% 

Category: insurance, Accuracy: 84.00% 

Category: finance, Accuracy: 70.00% 

Category: faq, Accuracy: 96.00%

256 50 

= VectorDatabase initialized = 

= Retriever initialized = 

= Ground truth accuracy = 

Retrieval accuracy: 84.00% 

Category: insurance, Accuracy: 90.00% 

Category: finance, Accuracy: 66.00% 

Category: faq, Accuracy: 96.00%

128 100 

= VectorDatabase initialized = 

= Retriever initialized = 

= Ground truth accuracy = 

Retrieval accuracy: 81.33% 

Category: insurance, Accuracy: 86.00% 

Category: finance, Accuracy: 62.00% 

Category: faq, Accuracy: 96.00%

1024 512 
 
Retrieval accuracy: 82.00% 

Category: insurance, Accuracy: 76.00% 

Category: finance, Accuracy: 74.00% 

Category: faq, Accuracy: 96.00%